google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineCloudConfigurable.java

```diff
@@ -86,16 +89,18 @@ public boolean isModified() {
   }
 
   /**
-   * User's shouldn't be able to save a cloud configuration without a valid Cloud SDK configured.
+   * Users shouldn't be able to save a cloud configuration without a valid Cloud SDK configured.
    */
   @Override
   public void apply() throws ConfigurationException {
-    if (StringUtil.isEmpty(cloudSdkPanel.getCloudSdkDirectory())
-        || !CloudSdkUtil.containsCloudSdkExecutable(cloudSdkPanel.getCloudSdkDirectory())) {
-      throw new RuntimeConfigurationError(
-          GctBundle.message("appengine.cloudsdk.location.invalid.message"));
-    } else {
+    try {
+      new CloudSdk.Builder()
+          .sdkPath(Paths.get(cloudSdkPanel.getCloudSdkDirectory()))
+          .build()
+          .validate();
       CloudSdkService.getInstance().setSdkHomePath(cloudSdkPanel.getCloudSdkDirectory());
+    } catch (AppEngineException aee) {
+      throw new RuntimeConfigurationError(aee.getMessage());
```

Comment:
I'd prefer a more specific exception vs a status code.

On Fri, Sep 2, 2016 at 10:29 AM, João Martins <notifications@github.com>
wrote:

> In google-cloud-tools-plugin/src/com/google/cloud/tools/
> intellij/appengine/cloud/AppEngineCloudConfigurable.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/857#discussion_r77355279>
> :
>
> >        CloudSdkService.getInstance().setSdkHomePath(cloudSdkPanel.getCloudSdkDirectory());
> > +    } catch (AppEngineException aee) {
> > +      throw new RuntimeConfigurationError(aee.getMessage());
>
> I thought about this too. We'd need to change AppEngineException to
> contain some sort of status code to present one of our messages. I can go
> ahead and implement this change in the common library.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/857/files/dc3870441d9aafd44df9d19e829c757914fea414#r77355279>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHf5HRj537J8bxZZfsB8rlfBY5HHxjW1ks5qmDLJgaJpZM4Jx_YN>
> .
>